front/src/app/modules/dashboard/components/dashboardOrganization/new-volunteering/new-volunteering.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, EventEmitter, Output } from '@angular/core';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { Store } from '@ngrx/store';
+import { DashboardServicesService } from '../../../services/dashboard-services.service';
+import { selectToken } from 'src/app/core/auth.selectors';
+
+@Component({
+  selector: 'app-new-volunteering',
+  templateUrl: './new-volunteering.component.html',
+  styleUrls: ['./new-volunteering.component.css'],
+})
+export class NewVolunteeringComponent {
+  @Output() declineNewVolunteering = new EventEmitter();
+  newVolunteering: FormGroup;
+  constructor(
+    private fb: FormBuilder,
+    private store: Store,
+    private dashServices: DashboardServicesService
+  ) {
+    this.newVolunteering = this.fb.group({
+      name: [''],
+      description: [''],
+      modeOfwork: [''],
+      workTime: [''],
+      vacancies: [0],
+      address: [''],
+      reward: [0],
+    });
+  }
+
+  onTimeWorkChange(keyForm: string, event: Event) {
+    const selectedValue = (event.target as HTMLSelectElement).value;
+    const updateObject = { [keyForm]: selectedValue };
+    this.newVolunteering.patchValue(updateObject);
+  }
+
+  closeNewVolunteering() {
+    this.declineNewVolunteering.emit();
+  }
+
+  addNewVolunteering() {
+    if (this.newVolunteering.valid) {
+      let formData = this.newVolunteering.value;
+      this.store.select(selectToken).subscribe((token) => {
+        if (token) {
```

Comment:
Hay formas mucho mejores para hacer el manejo del token. Personalmente recomiendo un Interceptor

https://jasonwatmore.com/post/2021/09/24/angular-http-interceptor-to-set-auth-header-for-api-requests-if-user-logged-in